docs/maps/index.rst

```diff
@@ -210,14 +216,134 @@ The following demonstrates how one can set pixel values:
+method argument.  The two spatial axes must always be named ``lon``
+and ``lat``.
+
+.. _mapcoord:
+
+MapCoord
+--------
+
+`~MapCoord` is an N-dimensional coordinate object that stores both
+spatial and non-spatial coordinates and is accepted by all ``coord``
+methods.  A `~MapCoord` can be created with or without explicitly
+named axes with `~MapCoord.create`.  Axes of a `~MapCoord` can be
+accessed by index, name, or attribute.  A `~MapCoord` without explicit
+axis names can be created by calling `~MapCoord.create` with a `tuple`
+argument:
+
+.. code:: python
+
+   import numpy as np
+   from astropy.coordinates import SkyCoord
+   from gammapy.maps import MapCoord
+
+   lon = np.array([0.0,1.0])
+   lat = np.array([1.0,2.0])
+   energy = np.array([100., 1000.])
+   skycoord = SkyCoord(lon, lat, unit='deg', frame='galactic')   
+
+   # Create a MapCoord from a tuple (no explicit axis names)
+   c = MapCoord.create((lon, lat, energy))
+   print(c[0], c['lon'], c.lon)
+   print(c[1], c['lat'], c.lat)
+   print(c[2], c['axis0'], c.axis0)
+
+   # Create a MapCoord from a tuple + SkyCoord (no explicit axis names)
+   c = MapCoord.create((skycoord, energy))
+   print(c[0], c['lon'], c.lon)
+   print(c[1], c['lat'], c.lat)
+   print(c[2], c['axis0'], c.axis0)
 
+The first two elements of the tuple argument must contain longitude
+and latitude.  Non-spatial axes are assigned a default name
+``axis{I}`` where ``{I}`` is the index of the non-spatial dimension.
+`~MapCoord` objects created without named axes must have the same axis
+ordering as the map geometry.
+
+
+A `~MapCoord` with named axes can be created by calling
+`~MapCoord.create` with a `dict` or `~collections.OrderedDict`:
+
+.. code:: python
+   
+   # Create a MapCoord from a dict
+   c = MapCoord.create(dict(lon=lon, lat=lat, energy=energy))
+   print(c[0], c['lon'], c.lon)
+   print(c[1], c['lat'], c.lat)
+   print(c[2], c['energy'], c.energy)
+
+   # Create a MapCoord from an OrderedDict
+   from collections import OrderedDict
+   c = MapCoord.create(OrderedDict([('energy',energy), ('lon',lon), ('lat', lat)]))
+   print(c[0], c['energy'], c.energy)
+   print(c[1], c['lon'], c.lon)
+   print(c[2], c['lat'], c.lat)
+   
+   # Create a MapCoord from a dict + SkyCoord
+   c = MapCoord.create(dict(skycoord=skycoord, energy=energy))
+   print(c[0], c['lon'], c.lon)
+   print(c[1], c['lat'], c.lat)
+   print(c[2], c['energy'], c.energy)
+
+
+Spatial axes must be named ``lon`` and ``lat``.  `~MapCoord` objects
+created with named axes do not need to have the same axis ordering as
+the map geometry.  However the name of the axis must match the name of
+the corresponding map geometry axis.
+
+   
 Interpolation
 -------------
 
-Maps support interpolation via the `~Map.get_by_coords` and
-`~Map.get_by_pix` methods.  Currently the following interpolation
+Maps support interpolation via the `~Map.interp_by_coord` and
+`~Map.interp_by_pix` methods.  Currently the following interpolation
```

Comment:
This link to `Map.interp_by_pix` doesn't resolve from the docs.
@woodmd - What is the thing to do here?
Add an abstract method to the base class?

Reply:
Yes having a base class method probably makes the most sense and would be parallel to what we have for the other accessor methods.  Adding this now.